pkg/repo/repo.go

```diff
@@ -112,6 +112,7 @@ func (r *File) Remove(name string) bool {
 
 // WriteFile writes a repositories file to the given path.
 func (r *File) WriteFile(path string, perm os.FileMode) error {
+	r.Generated = time.Now()
```

Comment:
If you use `NewFile` to create the `File` than `Generated` is set at that time. This is why using `helm repo index [directory]` has a generated time. To get `Generated` with a value of `'0001-01-01T00:00:00Z'` is means you are creating your own `File` instance without setting `Generated.

In this case, the new `Generated` value will overwrite the existing value if `NewFile` was used. We do not want that.

Can you share your use case and how your are encountering the problem.

Reply:
Thank you for your reply.
I just think that when adding a repo, there should be a default generation time. But based on what you said, I also think that the time should not be generated here, it should be before calling the `WriteFile` method. What's your point of view?